Bottom: 6c30bc51c82b267038ff63a398e6518d7ed6db61
Top:    0aee08b9ccb8fc0226b456d52e61f343d4e4999a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-06-05 16:08:53 +0100

Refresh of support-holes-in-input-polygon

---

diff --git a/src/crosshair.c b/src/crosshair.c
index e8542e0..a56a2ed 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -93,7 +93,7 @@ static void XORDrawAttachedArc (BDimension);
 static void
 XORPolygon (PolygonTypePtr polygon, LocationType dx, LocationType dy)
 {
-  Cardinal i
+  Cardinal i;
   for (i = 0; i < polygon->PointN; i++)
     {
       Cardinal next = next_contour_point (polygon, i);
@@ -473,13 +473,17 @@ XORDrawMoveOrCopyObject (void)
 
     case POLYGONPOINT_TYPE:
       {
-	PolygonTypePtr polygon = (PolygonTypePtr) Crosshair.AttachedObject.Ptr2;
-	PointTypePtr point = (PointTypePtr) Crosshair.AttachedObject.Ptr3;
-	Cadinal point_idx = point_idx = polygon_point_idx (polygon, point);
+	PolygonTypePtr polygon;
+	PointTypePtr point;
+	Cardinal point_idx, prev, next;
+
+	polygon = (PolygonTypePtr) Crosshair.AttachedObject.Ptr2;
+	point = (PointTypePtr) Crosshair.AttachedObject.Ptr3;
+	point_idx = polygon_point_idx (polygon, point);
 
 	/* get previous and following point */
-	Cardinal prev = prev_contour_point (polygon, point_idx);
-	Cardinal next = next_contour_point (polygon, point_idx);
+	prev = prev_contour_point (polygon, point_idx);
+	next = next_contour_point (polygon, point_idx);
 
 	/* draw the two segments */
 	gui->draw_line (Crosshair.GC,
